product_quick_bom/views/product_view.xml

```diff
@@ -7,7 +7,13 @@
             <field name="arch" type="xml">
                 <xpath expr="//page[@name='sales']" position="after">
                     <page name="bom_line" string="Bom line">
-                        <field name="bom_line_ids">
+                        <field name="display_bom_lines" attrs="{'invisible': 1}"/>
+                        <div attrs="{'invisible': [('display_bom_lines', '=', True)]}"
```

Comment:
I think you can avoid creating the field display_bom_lines and simply set {'invisible': [('bom_count', '<', 2)]}.

You just need to not add again the bom_count inside the smart button

Reply:
Usage of bom_count was not possible, because "product.product_template_form_view" (which impacts both product.product and product.template doesn't have bom_count. Now, I've extended product- and template- specific views and it works. Thanks